### PR TITLE
Merge pull requests on successful commit statuses

### PR DESCRIPTION
**Pull request by seemethere**
Basically mimics the `LGTM on green`.

When a commit status event comes through with the `success` state the
bot will go through and attempt to see if any PR's with that specific
SHA should be merged.

Also does checks like if the HEAD sha is the same as the SHA that the
commit status is for, and won't continue if there is a mismatch.

Quite a bit of refactoring in terms of abstracting away the
mergePullRequest method to allow for it to be used in both the context
of commit statuses as well as pull request review events.

Relates to #8

### CODE REVIEW

File: internal/webhook.go [hunk 1]
```diff
@@ -167,33 +184,90 @@ func GrabConfig(ctx context.Context, client *github.Client, repo, owner string,
...
 	if err != nil {
 		log.Fatal(err)
 	}
-	client := github.NewClient(&http.Client{Transport: itr})
+	return github.NewClient(&http.Client{Transport: itr})
+}
+
+func handleStatus(integrationID int, keyfile string, e github.StatusEvent) {
+	if *e.State != "success" {
+		log.Debugf("Skipping un-'successful'commit status event %s", *e.TargetURL)
```
Comment (jose-bigio) [inline internal/webhook.go]: Can this say not a successful commit status event?

**Review by corbin-coleman — APPROVED** [review]
LGTM

File: internal/webhook.go [hunk 2]
```diff
@@ -167,33 +184,90 @@ func GrabConfig(ctx context.Context, client *github.Client, repo, owner string,
...
 		keyfile,
 	)
 	if err != nil {
 		log.Fatal(err)
 	}
-	client := github.NewClient(&http.Client{Transport: itr})
+	return github.NewClient(&http.Client{Transport: itr})
+}
+
+func handleStatus(integrationID int, keyfile string, e github.StatusEvent) {
```
Comment (jose-bigio) [inline internal/webhook.go]: Could you name e something like `status_event` or `event`? I think it makes it more readable.

File: internal/webhook.go [hunk 3]
```diff
@@ -167,33 +184,90 @@ func GrabConfig(ctx context.Context, client *github.Client, repo, owner string,
 	return config, nil
 }
 
-// TODO: Write a function to save from people editing `.unir.yml` and auto-merging it
```
Comment (jose-bigio) [inline internal/webhook.go:170]: Could you add some element of the TODO above the editingConfig function to provide a description of what it's doing?

File: internal/webhook.go [hunk 4]
```diff
@@ -48,19 +48,26 @@ func (handler *GithubWebhookHandler) handleGithubWebhook(w http.ResponseWriter,
...
 	case *github.PullRequestReviewEvent:
 		go handlePullRequestReview(handler.integrationID, handler.keyfile, *e)
+	case *github.StatusEvent:
+		go handleStatus(handler.integrationID, handler.keyfile, *e)
 	}
 }
 
-func getPullRequestReviews(ctx context.Context, client *github.Client, e github.PullRequestReviewEvent) ([]*github.PullRequestReview, error) {
-	log.Debugf("[%s] Pulling pull request reviews", *e.Review.HTMLURL)
+func getPullRequestReviews(
```
Comment (jose-bigio) [inline internal/webhook.go:56]: Can you put function declarations on one line? 

https://github.com/google/go-github/blob/master/github/pulls_reviews.go#L69

**Review by andrewhsu — APPROVED** [review]
LGTM wrt having the ability to merge on green; can this be configurable?

**Comment by seemethere** [comment]
@andrewhsu it could be configurable, I can add that in as a follow up PR

File: internal/webhook.go [hunk 5]
```diff
@@ -167,33 +184,90 @@ func GrabConfig(ctx context.Context, client *github.Client, repo, owner string,
...
+			client,
+			*e.Repo.Owner.Login,
+			*e.Repo.Name,
+			*pull_request.Head.SHA,
+			*pull_request.Number,
+		)
+	}
+}
+
+// TODO: Write a function to save from people editing `.unir.yml` and auto-merging it
```
Comment (jose-bigio) [inline internal/webhook.go]: Can you delete this since it's already implemented?

File: internal/webhook.go [hunk 6]
```diff
@@ -167,33 +184,90 @@ func GrabConfig(ctx context.Context, client *github.Client, repo, owner string,
...
+	client := createGithubClient(integrationID, *e.Installation.ID, keyfile)
 	log.Debugf("[%s] STARTED handling pull request review", *e.Review.HTMLURL)
+	mergePullRequest(client, *e.Repo.Owner.Login, *e.Repo.Name, *e.PullRequest.Head.SHA, *e.PullRequest.Number)
+}
+
+func mergePullRequest(client *github.Client, owner, repo, sha string, pr_number int) {
 	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
 	defer cancel()
-	allReviews, err := getPullRequestReviews(ctx, client, e)
+	githubURL := fmt.Sprintf("https://github.com/%s/%s/pull/%d", owner, repo, pr_number)
```
Comment (jose-bigio) [inline internal/webhook.go]: Probably won't change, but github doesn't guarantee these urls won't change.

**Review by jose-bigio — CHANGES_REQUESTED** [review]
A lot of the comments are style based

File: internal/webhook.go [hunk 7]
```diff
@@ -167,33 +184,86 @@ func GrabConfig(ctx context.Context, client *github.Client, repo, owner string,
...
 	)
 	if err != nil {
 		log.Fatal(err)
 	}
-	client := github.NewClient(&http.Client{Transport: itr})
-	log.Debugf("[%s] STARTED handling pull request review", *e.Review.HTMLURL)
+	return github.NewClient(&http.Client{Transport: itr})
+}
+
+func handleStatus(integrationID int, keyfile string, statusEvent github.StatusEvent) {
```
Comment (jose-bigio) [inline internal/webhook.go:200]: statusEvent is camel case, but other variables are snake case.

**Review by jose-bigio — APPROVED** [review]
LGTM

**Review by jose-bigio — APPROVED** [review]
LGTM

**Review by corbin-coleman — APPROVED** [review]
LGTM

**Pull request merged**